OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -70,13 +70,17 @@ public class HarvesterInfo : ITraitInfo, Requires<MobileInfo>
 		[Desc("The pathfinding cost penalty applied for each harvester waiting to unload at a refinery.")]
 		public readonly int UnloadQueueCostModifier = 12;
 
+		[GrantedConditionReference]
+		[Desc("Condition to grant while empty.")]
+		public readonly string EmptyCondition = null;
```

Comment:
Would be cool if we can define values here. We could drop whatever that handles the empty-half-full artwork to use conditions instead too.

There is what i used for Generals Alpha for supply logic: https://github.com/MustaphaTR/OpenRA/blob/generals-alpha-engine/OpenRA.Mods.Gen/Traits/Supply/SupplyCollector.cs#L61 https://github.com/MustaphaTR/OpenRA/blob/generals-alpha-engine/OpenRA.Mods.Gen/Traits/Supply/SupplyDock.cs#L37
Here we could use bale amount instead, since value depends on what is in.

Reply:
The whole bale system needs to be rethought as part of a wider player resource rework. I think it would be a mistake to further bake in what we already know are bad abstractions now.